src/toxcore_tests/dht_tests.rs

```diff
@@ -816,7 +817,7 @@ fn get_nodes_to_bytes_test() {
         let g_bytes = gn.to_bytes();
         let PublicKey(pk_bytes) = gn.pk;
         assert_eq!(&pk_bytes, &g_bytes[..PUBLICKEYBYTES]);
-        assert_eq!(&u64_to_array(gn.id), &g_bytes[PUBLICKEYBYTES..]);
+        assert_eq!(gn.id, LittleEndian::read_u64(&g_bytes[PUBLICKEYBYTES..]));
```

Comment:
I meant I convert now the right part to u64 but there was conversion of left part to array. With byteorder you can't convert integer to array with static size.
About endianess - why here should be NativeEndian? You used here u64_to_array - it converts from LittleEndian always.